flipperdevices/flipperzero-firmware
# [FL-3662] Do not remove file when renaming to itself

# What's new

- Enchanced checks for `storage_common_remove()` function.
- Add `storage_common_equivalent_path()` function.
- Renaming a file to itself always succeeds.
- Renaming a directory to itself or to a subdirectory of itself always fails.
- Overwriting a regular file with a directory always fails.
- Per-storage support for different case behaviour.
- Update `storage.h` docs.

# Verification 

1. Access Flipper's filesystem via an RPC client.
2. Try renaming a directory to the name of an already existing file. The operation should fail with `FSE_INVALID_NAME` error.
3. Repeat step 2, but change the case of some characters in the name. The result should be the same.
4. Try renaming a directory to the same name. The operation should fail with `FSE_INVALID_NAME` error.
5. Same as step 3, but in relation to step 4.
6. Try renaming a file to the same name. The operation should succeed.
7. Same as step 3, but in relation to step 6.

# Checklist (For Reviewer)
